GradeBook/GradeBook.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GradeBook
+{
+    class Program
+    {
+        static void Main(string[] args)
```

Comment:
I like the the review functionality, and you logic is correct. But you should consider splitting all of this logic up into different methods. Having it all in one method makes it hard to follow.